### PR TITLE
proposal(bep): socks support for HTTP package

### PR DESCRIPTION
**Pull request by ThisaruGuruge**
## Purpose

This is the proposal for https://github.com/ballerina-platform/ballerina-spec/issues/1450

## SOCKS Proxy Support Enhancement Proposal

Adds a new Ballerina Enhancement Proposal (BEP) documenting extension of the HTTP package’s client proxy capabilities to support socket-level SOCKS proxies (SOCKS4 and SOCKS5), enabling enterprise-style egress setups that rely on SOCKS rather than HTTP proxies.

The proposal introduces:
- **API changes**: Extend `http:ProxyConfig` with a `protocol` discriminator and a new `ProxyProtocol` enum (`HTTP` default, `SOCKS4`, `SOCKS5`).
- **Protocol-specific behavior**: Clarifies how SOCKS4/SOCKS5 differ in authentication support (SOCKS5 username/password, SOCKS4 `userId` only) and DNS resolution expectations for SOCKS5 (proxy-side/remote DNS).
- **Validation & configuration**: Specifies input validation rules (e.g., rejecting unsupported SOCKS4 password configuration).
- **Java/Netty integration design**: Describes wiring via Netty SOCKS proxy handlers, including resolver behavior scoping to avoid affecting non-SOCKS paths.
- **Testing plan**: Covers Java unit tests, Ballerina unit tests (including SOCKS4 negative validation), and Ballerina integration tests, using an embedded Netty SOCKS server for end-to-end verification.

**File Changes**: +180 lines (1 new documentation file): `beps/lib-http/1450_socks_support.md`

### CODE REVIEW

File: beps/lib-http/1450_socks_support.md [hunk 1]
```diff
@@ -0,0 +1,183 @@
+# NNNN: Socket-Level (SOCKS) Proxy Support for the HTTP Client
```
Comment (TharmiganK) [inline beps/lib-http/1450_socks_support.md]:
```suggestion
# 1450: Socket-Level (SOCKS) Proxy Support for the HTTP Client
```

File: beps/lib-http/1450_socks_support.md [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+
+- Authors - Thisaru Guruge
+- Reviewed by - TBD
+- Created date - 2026-06-18
+- Updated date - 2026-06-18
+- Issue - https://github.com/wso2-enterprise/integration-product-management/issues/850
```
Comment (TharmiganK) [inline beps/lib-http/1450_socks_support.md]: Link the spec issue instead

**Review by daneshk — APPROVED** [review]
